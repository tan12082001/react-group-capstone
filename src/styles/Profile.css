.profile-outer {
  display: grid;
  grid-template-columns: 49% 49%;
  gap: 2%;
  padding: 3rem;
  height: 100%;
}

.filtered-missions {
  display: flex;
  flex-direction: column;
  gap: 1rem;
}

.mission-names {
  border: 1px solid gray;
}

.each-mission-name {
  border-bottom: 1px solid gray;
  padding: 1.6rem;
}
