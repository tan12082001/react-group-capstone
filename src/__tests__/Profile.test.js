import React from 'react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { screen, render } from '@testing-library/react';
import Profile from '../Components/MyProfilePage';
import store from '../redux/store';
import configureMockStore from 'redux-mock-store';
import FilteredMissions from '../Components/MissionComponents/FilterMissions';
// import FilteredRockets from '../Components/RocketComponents/FilterRockets';

const mock = configureMockStore([]);

describe('Profile page testing', () => {
  test('Check if MyProfile component is rendered correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Profile />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  test('Check for inital render', () => {
    render(<Provider store={store}><Profile /></Provider>);
    expect(screen.getByText('My Missions')).toBeVisible();
    expect(screen.getByText('No current missions')).toBeVisible();
    expect(screen.getByText('My Rockets')).toBeVisible();
    expect(screen.getByText('No rockets reserved')).toBeVisible();
  });
  test('check the filtered missions', () => {
    const store = mock({
      missions: {
        missions: [
          {
            missionName: 'Mission 1',
            missionJoin: true,
          },
          {
            missionName: 'Mission 2',
            missionJoin: false,
          },
          {
            missionName: 'Mission 3',
            missionJoin: true,
          }
        ]
      },

    })
    const checkt = render(<Provider store={store}><FilteredMissions /></Provider>);
    // expect(checkt).toMatchSnapshot(); this is also giving the same output as below lines.
    expect(checkt.getByText('Mission 1')).toBeVisible();
    expect(checkt.getByText('Mission 3')).toBeVisible();
  })
  /* test('check filtered rockets', () => {
    const store = mock({
      rockets: {
        rockets: [
          {
            name: 'Rocket 1',
            reserved: true,
          }
        ]
      }
    })
    render(<Provider store={store}><FilteredRockets /></Provider>);
    expect(screen.getByText('Rocket 1')).toBeVisible();
  }) */
});
