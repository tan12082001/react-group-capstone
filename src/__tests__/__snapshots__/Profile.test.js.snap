// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Case 1: MyProfile Page test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-outer"
      >
        <div
          class="filtered-missions"
        >
          <h2>
            My Missions
          </h2>
          <div
            class="each-mission-name"
          >
            No current missions
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-outer"
    >
      <div
        class="filtered-missions"
      >
        <h2>
          My Missions
        </h2>
        <div
          class="each-mission-name"
        >
          No current missions
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Check if MyProfile component is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-outer"
      >
        <div
          class="filtered-missions"
        >
          <h2>
            My Missions
          </h2>
          <div
            class="each-mission-name"
          >
            No current missions
          </div>
        </div>
        <div
          class="filtered-rockets"
        >
          <h2>
            My Rockets
          </h2>
          <div
            class="each-rocket-name"
          >
            No rockets reserved
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-outer"
    >
      <div
        class="filtered-missions"
      >
        <h2>
          My Missions
        </h2>
        <div
          class="each-mission-name"
        >
          No current missions
        </div>
      </div>
      <div
        class="filtered-rockets"
      >
        <h2>
          My Rockets
        </h2>
        <div
          class="each-rocket-name"
        >
          No rockets reserved
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MyProfile page Test case Case 1: MyProfile Page test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-outer"
      >
        <div
          class="filtered-missions"
        >
          <h2>
            My Missions
          </h2>
          <div
            class="each-mission-name"
          >
            No current missions
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-outer"
    >
      <div
        class="filtered-missions"
      >
        <h2>
          My Missions
        </h2>
        <div
          class="each-mission-name"
        >
          No current missions
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Profile page testing Check if MyProfile component is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-outer"
      >
        <div
          class="filtered-missions"
        >
          <h2>
            My Missions
          </h2>
          <div
            class="each-mission-name"
          >
            No current missions
          </div>
        </div>
        <div
          class="filtered-rockets"
        >
          <h2>
            My Rockets
          </h2>
          <div
            class="each-rocket-name"
          >
            No rockets reserved
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-outer"
    >
      <div
        class="filtered-missions"
      >
        <h2>
          My Missions
        </h2>
        <div
          class="each-mission-name"
        >
          No current missions
        </div>
      </div>
      <div
        class="filtered-rockets"
      >
        <h2>
          My Rockets
        </h2>
        <div
          class="each-rocket-name"
        >
          No rockets reserved
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Profile page testing check the missions 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filtered-missions"
      >
        <h2>
          My Missions
        </h2>
        <div
          class="mission-names"
        >
          <div
            class="each-mission-name"
          >
            Mission 1
          </div>
          <div
            class="each-mission-name"
          >
            Mission 3
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filtered-missions"
    >
      <h2>
        My Missions
      </h2>
      <div
        class="mission-names"
      >
        <div
          class="each-mission-name"
        >
          Mission 1
        </div>
        <div
          class="each-mission-name"
        >
          Mission 3
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
